code/modules/mob/living/simple_animal/hostile/carp.dm

```diff
@@ -146,7 +146,16 @@
 			user.drop_item(F, force_drop = 1)
 
 			if(prob(25))
-				if(!friends.Find(user))
+				if(!friends.Find(user)) //Not friends with the feeder yet
+
+					//The following three lines allow baby carps to start seeing ALL humans / vox / monkeys / etc as friends.
+					//If only one human feeds a carp, the carp will only see that human as a friend.
+					//If, however, two or more humans feed a carp, the carps will see the entire human species as friends. They'll still be agressive towards vox (unless two or more vox feed them) and other mobs.
+					for(var/mob/M in friends)
+						if(M.type == user.type)
+							friends[user.type] += 2
+							break
+
 					friends.Add(user)
```

Comment:
Uh
Carp still storing references to player mobs who will need to be qdel'd when they explode horribly = bad